cmd/maya-apiserver/app/server/backup_endpoint_v1alpha1.go

```diff
@@ -0,0 +1,388 @@
+			return &cvr
+		}
+	}
+
+	return nil
+}
+
+// getLastBackupSnap will fetch the last successful backup's snapshot name
+func getLastBackupSnap(openebsClient *internalclientset.Clientset, bkp *v1alpha1.BackupCStor) (string, error) {
+	lastbkpname := bkp.Spec.BackupName + "-" + bkp.Spec.VolumeName
+	b, err := openebsClient.OpenebsV1alpha1().BackupCStorLasts(bkp.Namespace).Get(lastbkpname, v1.GetOptions{})
+	if err != nil {
+		bk := &v1alpha1.BackupCStorLast{
+			ObjectMeta: v1.ObjectMeta{
+				Name:      lastbkpname,
+				Namespace: bkp.Namespace,
+				Labels:    bkp.Labels,
+			},
+			Spec: v1alpha1.BackupCStorSpec{
+				BackupName:   bkp.Spec.BackupName,
+				VolumeName:   bkp.Spec.VolumeName,
+				PrevSnapName: bkp.Spec.SnapName,
+			},
+		}
+
+		_, err := openebsClient.OpenebsV1alpha1().BackupCStorLasts(bk.Namespace).Create(bk)
+		if err != nil {
+			glog.Errorf("Error creating last-backup resource for backup:%v err:%v", bk.Spec.BackupName, err)
+			return "", err
+		}
+		glog.Infof("LastBackup resource created for backup:%s volume:%s", bk.Spec.BackupName, bk.Spec.VolumeName)
+		return "", nil
+	}
+	return b.Spec.PrevSnapName, nil
+}
+
+// get is http handler which handles backup get request
+func (bOps *backupAPIOps) get() (interface{}, error) {
+	bkp := &v1alpha1.BackupCStor{}
+
+	err := decodeBody(bOps.req, bkp)
+	if err != nil {
+		return nil, err
+	}
+
+	// backup name is expected
+	if len(strings.TrimSpace(bkp.Spec.BackupName)) == 0 {
+		return nil, CodedError(400, fmt.Sprintf("Failed to get backup: missing backup name "))
+	}
+
+	// namespace is expected
+	if len(strings.TrimSpace(bkp.Namespace)) == 0 {
+		return nil, CodedError(400, fmt.Sprintf("Failed to get backup '%v': missing namespace", bkp.Spec.BackupName))
+	}
+
+	// volume name is expected
+	if len(strings.TrimSpace(bkp.Spec.VolumeName)) == 0 {
+		return nil, CodedError(400, fmt.Sprintf("Failed to get backup '%v': missing volume name", bkp.Spec.BackupName))
+	}
+
+	openebsClient, k8sClient, err := loadClientFromServiceAccount()
+	if err != nil {
+		return nil, CodedError(400, fmt.Sprintf("Failed to create openEBSClient '%v'", err))
+	}
+
+	bkp.Name = bkp.Spec.SnapName + "-" + bkp.Spec.VolumeName
+	b, err := openebsClient.OpenebsV1alpha1().BackupCStors(bkp.Namespace).Get(bkp.Name, v1.GetOptions{})
+	if err != nil {
+		return nil, CodedError(400, fmt.Sprintf("Failed to fetch backup error:%v", err))
+	}
+
+	if b.Status != v1alpha1.BKPCStorStatusDone && b.Status != v1alpha1.BKPCStorStatusFailed {
+		// check if node is running or not
+		bkpNodeDown := checkIfBKPPoolNodeDown(k8sClient, b)
+		// check if cstor-pool-mgmt container is running or not
+		bkpPodDown := checkIfBKPPoolPodDown(k8sClient, b)
+
+		if bkpNodeDown || bkpPodDown {
+			// Backup is stalled, let's find last-backup status
+			laststat := findLastBackupStat(openebsClient, b)
+			// Update Backup status according to last-backup
+			updateBackupStatus(openebsClient, b, laststat)
```

Comment:
job to update backupcstor is of the pool-mgmt not the api server.
and, API server should just pick the status from backupcstor of that schedule

Reply:
got the point.. thanks for explaining. More comments would be helpful and update in the md file would be great

Reply:
ok